captum/attr/_utils/common.py

```diff
@@ -238,16 +372,52 @@ def _format_and_verify_sliding_window_shapes(sliding_window_shapes, inputs):
     return sliding_window_shapes
 
 
+@typing.overload
+def _compute_conv_delta_and_format_attrs(
+    attr_algo: "GradientAttribution",
+    return_convergence_delta: bool,
+    attributions: Tuple[Tensor, ...],
+    start_point: Union[int, float, Tensor, Tuple[Union[int, float, Tensor], ...]],
+    end_point: Union[Tensor, Tuple[Tensor, ...]],
+    additional_forward_args: Any,
+    target: Optional[
+        Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]], List[int]]
+    ],
+    is_inputs_tuple: "Literal"[False] = False,
+) -> Union[Tensor, Tuple[Tensor, Tensor]]:
+    ...
+
+
+@typing.overload
+def _compute_conv_delta_and_format_attrs(
+    attr_algo: "GradientAttribution",
+    return_convergence_delta: bool,
+    attributions: Tuple[Tensor, ...],
+    start_point: Union[int, float, Tensor, Tuple[Union[int, float, Tensor], ...]],
+    end_point: Union[Tensor, Tuple[Tensor, ...]],
+    additional_forward_args: Any,
+    target: Optional[
+        Union[int, Tuple[int, ...], Tensor, List[Tuple[int, ...]], List[int]]
+    ],
+    is_inputs_tuple: "Literal"[True],
+) -> Union[Tuple[Tensor, ...], Tuple[Tuple[Tensor, ...], Tensor]]:
+    ...
+
+
 def _compute_conv_delta_and_format_attrs(
-    attr_algo,
-    return_convergence_delta,
-    attributions,
-    start_point,
-    end_point,
-    additional_forward_args,
-    target,
-    is_inputs_tuple=False,
-):
+    attr_algo: "GradientAttribution",
```

Comment:
Overloads are only used to identify signature for callers of a method, but the method implementation is not type checked unless the implementation is type hinted, so adding type hints to all method implementations as well.

Reply:
I see. Let's add a #FIXME for this so that we can fix the circular import problem when we refactor `common.py`